File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> map;
        vector<int> result(2);
        for (int i = 0; i < nums.size(); ++i) {
            if (map.find(target - nums[i]) == map.end()) {
                map[nums[i]] = i;
            } else {
                result[1] = i;
                result[0] = map[target - nums[i]];
                return result;
            }
        }
        return result;
    }
};

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] result = new int[2];
        Map map = new HashMap();
        for (int i = 0; i < nums.length; ++i) {
            if (!map.containsKey(target - nums[i])) {
                map.put(nums[i], i);
            } else {
                result[1] = i;
                result[0] = (int) map.get(target - nums[i]);
                return result;
            } 
        }
        return result;
    }
}